src/main/java/seedu/duke/Inventory.java

```diff
@@ -232,4 +231,35 @@ public static void listItems() {
             Ui.printInvalidList();
         }
     }
+
+    /**
+     * Removes item from the array list of items and hashmap of upc
+     * using item's unique upc.
+     * @param upc UPC of item to be removed from the list.
+     * @return index of item in the arraylist for testing; otherwise -1.
+     */
+    public static int removeByUpc(String upc) {
+        String upcCode = upc.replaceFirst("upc/", "");
+        searchUPC(upcCode);
+        Ui.printConfirmMessage();
+        Scanner s = new Scanner(System.in);
+        String confirmation = s.nextLine();
+        switch(confirmation) {
+        case "Y":
+            Item itemToRemove = upcCodes.get(upcCode);
+            int i = items.indexOf(itemToRemove);
+            upcCodes.remove(upcCode);
+            items.remove(i);
+            listItems();
+            return i;
```

Comment:
Try to integrate with trie and the hashmap if possible (see removeItemAtIndex for more info). If not I can just do it after merge.

Reply:
by "the hashmap" I mean itemNameHash